### PR TITLE
JR2-148 add suggest support for locality autocomplete

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/reference/service/dto/LocalitySuggestionDto.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package ch.admin.seco.service.reference.service.dto;
+
+public class LocalitySuggestionDto {
+
+    private String code;
+
+    private String name;
+
+    public LocalitySuggestionDto(String code, String name) {
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/reference/service/dto/LocalitySuggestionDto.java]: What is code?

Add `communalCode` and `cantonCode`

File: src/main/java/ch/admin/seco/service/reference/domain/Locality.java [hunk 2]
```diff
@@ -71,6 +76,11 @@
     @Embedded
     private GeoPoint geoPoint;
 
+    @Transient
+    @JsonProperty("citySuggestions")
+    @JsonView(LocalityViews.ElasticSearch.class)
```
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/reference/domain/Locality.java]: create a dedicated entity for the elasticsearch. (if it's just an additional field you can also extend it)

the @JsonView boilerplate is difficult to understand and makes the code more complex. the models should be clean from this. The view is to mask the data the user can see, but should not be used to modify data models

Maybe later in the project we're coming back to this solution when we need to return different sets of data depending on the user role.

Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/domain/Locality.java]: Created LocalitySynonym for ES indexing

File: src/main/java/ch/admin/seco/service/reference/service/dto/LocalitySuggestionDto.java (hunk 1 above)
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/service/dto/LocalitySuggestionDto.java]: Renamed code id and added `communalCode` and `cantonCode`

File: src/main/java/ch/admin/seco/service/reference/config/JacksonConfiguration.java [hunk 3]
```diff
@@ -9,9 +10,9 @@
 @Configuration
 public class JacksonConfiguration {
 
-    /*
-     * Support for Hibernate types in Jackson.
-     */
+    public JacksonConfiguration(ObjectMapper objectMapper) {
```
Comment (varusakov) [inline src/main/java/ch/admin/seco/service/reference/config/JacksonConfiguration.java]: Revert this changes back

File: src/main/resources/config/elasticsearch/mappings/locality.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
             "geoPoint": {
                 "type": "geo_point"
             },
-            "citySuggestions": {
+            "suggestions": {
```
Comment (michael-wirth) [inline src/main/resources/config/elasticsearch/mappings/locality.json]: Because of a limitation of elasticsearch, the suggest feature can't search by "type". I will search for all field called "suggestions" within the same index

There will also be the search for canton, where we have to match the name or the cantonCode, which might also be problematic. 

We will solve it when we implement the autocomplete for canton

**Comment by varusakov** [comment]
Updated LocalityGatlingTest and rebase on `develop` branch
